code/API_definitions/population-density-data.yaml

```diff
@@ -0,0 +1,525 @@
+            and must have time zone. Recommended format is yyyy-MM-dd'T'HH:mm:ss.SSSZ
+            (i.e. which allows 2023-07-03T14:27:08+02:00 or 2023-07-03T12:27:08Z).
+          example: '2024-01-03T13:00:00Z'
+        populationData:
+          $ref: '#/components/schemas/PopulationData'
+      required:
+        - startTime
+        - endTime
+        - populationData
+    PopulationData:
+      description: >-
+        Object that contains the maximum, minimum and average population in a
+        cell for a specific interval. In case of insufficient data to guarantee an
+        anonymised prediction due to the k-anonymity within a specific cell and
+        time range, no population data will be returnet and the property dataType
+        value will be "LOW_DENSITY".
+      properties:
+        dataType:
+          type: string
+          enum:
+            - LOW_DENSITY
+            - DENSITY_ESTIMATION
+      required:
+        - dataType
+      discriminator:
+        propertyName: dataType
+        mapping:
+          LOW_DENSITY: '#/components/schemas/PopulationData'
+          DENSITY_ESTIMATION: '#/components/schemas/DensityEstimationPopulationData'
+    DensityEstimationPopulationData:
+      allOf:
+        - $ref: '#/components/schemas/PopulationData'
+        - type: object
+          properties:
+            maxPopulation:
+              type: number
+              format: double
+              description: Maximum population expected in the defined area.
+            minPopulation:
+              type: number
+              format: double
+              description: Minimum population expected in the defined area.
+            avgPopulation:
+              type: number
+              format: double
+              description: Average population expected in the defined area.
+          required:
+            - maxPopulation
+            - minPopulation
+            - avgPopulation
+    ErrorInfo:
+      type: object
+      required:
+        - status
+        - code
+        - message
+      properties:
+        status:
+          type: integer
+          description: HTTP status code returned along with this error response.
+        code:
+          type: string
+          description: Code given to this error.
+        message:
+          type: string
+          description: Detailed error description.
+  responses:
+    RetrieveLocationBadRequest400:
+      description: >-
+        Problem with the client request. In addition to regular scenario of
+        `INVALID_ARGUMENT`, another scenarios may exist:
+          - The area is not a polygon shape or has an arbitrary complexity ("code": "POPULATION_DENSITY_DATA.INVALID_AREA", "message": "The area is not a polygon shape or has an arbitrary complexity")
+          - The area or part of the area is outside the allowed area ("code": "POPULATION_DENSITY_DATA.OUTSIDE_ALLOWED_AREA", "message": " The area or part of the area is outside the allowed area")

```

Comment:
If a "part of the area is outside the allowed area", are we considering the water zone area ?